### PR TITLE
Removed copyWith attribute in SearchDelegate #32180

### LINKED ISSUE #32180: Search text on showSearch with dark theme is unreadable
image

Simply using ThemeData.dark();

### PR DESCRIPTION
**Pull request by JoseAlba**
## Description

Removed copyWith attribute in SearchDelegate because it was forcing users to use those attributes. This would cause problems when users wanted to create a custom dark search delegate.

## Related Issues

Closes #32180 

## Tests

I added the following tests:

Added test to search_test.dart to see if you can change the colour of the search bar.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/search_demo.dart [hunk 1]
```diff
@@ -60,7 +60,7 @@ class _SearchDemoState extends State<SearchDemo> {
                   ? Icons.more_horiz
                   : Icons.more_vert,
             ),
-            onPressed: () { },
+            onPressed: () {},
```
Comment (rami-a) [inline examples/flutter_gallery/lib/demo/material/search_demo.dart]: Looks like you have formatting enabled, you should turn it off and undo all the formatting changes in all these files.

File: packages/flutter/test/material/search_test.dart [hunk 2]
```diff
@@ -100,11 +103,27 @@ void main() {
...
     await tester.pumpAndSettle();
 
-    final TextField textField = tester.widget<TextField>(find.byType(TextField));
+    final TextField textField =
+        tester.widget<TextField>(find.byType(TextField));
     final Color hintColor = textField.decoration.hintStyle.color;
     expect(hintColor, delegate.hintTextColor);
   });
 
+  testWidgets('SearchBar color overridden', (WidgetTester tester) async {
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: I tried to run this test with the old search code and it passes there (when I would expect it to fail), so I think you may have to re-work the test somehow so that it fails with the existing code and passes with your changes

File: packages/flutter/lib/src/material/search.dart [hunk 3]
```diff
@@ -193,12 +192,7 @@ abstract class SearchDelegate<T> {
     assert(context != null);
     final ThemeData theme = Theme.of(context);
     assert(theme != null);
-    return theme.copyWith(
-      primaryColor: Colors.white,
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/search.dart]: is there a way to keep the copyWith here but use the correct colors depending on light/dark theme?

**Comment by dmytrokravcuk** [comment]
Pomoc

File: packages/flutter/test/material/search_test.dart (hunk 2 above)
Comment (JoseAlba) [reply packages/flutter/test/material/search_test.dart]: Done, made a test that fails without the changes

File: packages/flutter/test/material/search_test.dart [hunk 4]
```diff
@@ -659,6 +673,9 @@ class TestHomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      theme: ThemeData(
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Undo this change

File: packages/flutter/test/material/search_test.dart [hunk 5]
```diff
@@ -618,7 +634,6 @@ void main() {
 
       expect(semantics, hasSemantics(buildExpected(routeName: 'Search'),
           ignoreId: true, ignoreRect: true, ignoreTransform: true));
-
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Undo this change

File: packages/flutter/test/material/search_test.dart [hunk 6]
```diff
@@ -635,7 +650,6 @@ void main() {
 
       expect(semantics, hasSemantics(buildExpected(routeName: ''),
           ignoreId: true, ignoreRect: true, ignoreTransform: true));
-
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Undo this change

File: packages/flutter/test/material/search_test.dart [hunk 7]
```diff
@@ -101,10 +101,26 @@ void main() {
     await tester.pumpAndSettle();
 
     final TextField textField = tester.widget<TextField>(find.byType(TextField));
+
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Undo this change

File: packages/flutter/lib/src/material/search.dart [hunk 8]
```diff
@@ -8,14 +8,14 @@ import 'package:flutter/foundation.dart';
...
 import 'app_bar.dart';
-import 'colors.dart';
 import 'debug.dart';
 import 'input_border.dart';
 import 'input_decorator.dart';
 import 'material_localizations.dart';
 import 'scaffold.dart';
 import 'text_field.dart';
 import 'theme.dart';
+import 'colors.dart';
```
Comment (rami-a) [inline packages/flutter/lib/src/material/search.dart]: Undo this change

File: examples/flutter_gallery/lib/demo/material/search_demo.dart [hunk 9]
```diff
@@ -153,7 +153,6 @@ class _SearchDemoSearchDelegate extends SearchDelegate<int> {
 
   @override
   Widget buildSuggestions(BuildContext context) {
-
```
Comment (rami-a) [inline examples/flutter_gallery/lib/demo/material/search_demo.dart]: Undo this change

File: packages/flutter/test/material/search_test.dart [hunk 10]
```diff
@@ -751,3 +768,56 @@ class _TestSearchDelegate extends SearchDelegate<String> {
     return actions;
   }
 }
+
+class _TestSearchDelegate2 extends SearchDelegate<String> {
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Rename this class to `_TestSearchDelegatePrimaryColor` or something like that

File: packages/flutter/test/material/search_test.dart [hunk 11]
```diff
@@ -751,3 +768,56 @@ class _TestSearchDelegate extends SearchDelegate<String> {
...
+  _TestSearchDelegate2({
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Remove everything from this class if it isn't needed, like all the constructors, properties, overrides.

File: packages/flutter/lib/src/material/search.dart (hunk 8 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/search.dart]: I removed it since it is not needed anymore

File: packages/flutter/test/material/search_test.dart (hunk 4 above)
Comment (JoseAlba) [reply packages/flutter/test/material/search_test.dart]: This is needed for the test

File: packages/flutter/test/material/search_test.dart (hunk 10 above)
Comment (JoseAlba) [reply packages/flutter/test/material/search_test.dart]: Done

File: packages/flutter/test/material/search_test.dart [hunk 12]
```diff
@@ -751,3 +769,30 @@ class _TestSearchDelegate extends SearchDelegate<String> {
     return actions;
   }
 }
+
+class _TestSearchDelegatePrimaryColor extends SearchDelegate<String> {
+  _TestSearchDelegatePrimaryColor();
+
+  final Color primaryColor = Colors.pink;
```
Comment (rami-a) [inline packages/flutter/test/material/search_test.dart]: Move this into the actual test since the delegate isn't providing this

**Review by rami-a — APPROVED** [review]

File: packages/flutter/test/material/search_test.dart (hunk 12 above)
Comment (JoseAlba) [reply packages/flutter/test/material/search_test.dart]: Done

Comment (JoseAlba) [reply packages/flutter/test/material/search_test.dart]: Done

**Review by dmytrokravcuk — APPROVED** [review]

**Comment by JoseAlba** [comment]
Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 14 10 47

**Comment by HansMuller** [comment]
The documented behavior for the SearchDelegate's [appBarTheme](https://api.flutter.dev/flutter/material/SearchDelegate/appBarTheme.html) method is that it provides a theme with a white background. It's been that way for about 2 years, https://github.com/flutter/flutter/pull/17629. Changing it would require tests/doc and following the [breaking change process](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes).

**Comment by HansMuller** [comment]
I'm going to close this per https://github.com/flutter/flutter/pull/48830#issuecomment-575267114

If you plan to continue working on it, please reopen the PR when you're ready.
